### PR TITLE
Add flake (fix #7), update Dyalog and RIDE, fix package names in RIDE

### LINKED ISSUE #7: Implement a flake
Making a flake would allow running `nix run github:markus1189/dyalog-nixos` while getting an environment to run dyalog APL, which would be rather convenient.

### PR DESCRIPTION
**Pull request by MaxCan-Code**

### CODE REVIEW

**Comment by markus1189** [comment]
Hey @MaxCan-Code thank you for the pr!  Does the new RIDE version start successfully for you?  I only get the splash screen and nothing happens...

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  inputs.ride.url = github:Dyalog/ride;
+  inputs.ride.flake = false;
```
Comment (markus1189) [inline flake.nix:3]: Are these inputs ever used?

**Comment by MaxCan-Code** [comment]
Same here. I'm no expert in electron apps but I tried a few fixes. None worked so I just left it as is. I've resorted to using `RIDE_INIT=http:*:8888 dyalog` and connecting to RIDE at http://localhost:8888

File: flake.nix (hunk 1 above)
Comment (MaxCan-Code) [reply flake.nix:3]: Not yet. It's groundwork for accessing RIDE's source code more conveniently

Comment (MaxCan-Code) [reply flake.nix:3]: Someone else can build onto that work. I don't think I'll have the time

**Comment by markus1189** [comment]
Thanks a lot for the flake!

**Pull request merged**